test_util/cluster_api.py

```diff
@@ -227,11 +223,10 @@ def _authenticate_default_user(self):
         will raise exception if authorization fails
         """
         self.web_auth_default_user.authenticate(self)
-        self.default_headers.update(self.web_auth_default_user.auth_header)
+        self.session.auth = self.web_auth_default_user.auth
 
     def __init__(self, dcos_url, masters, public_masters, slaves, public_slaves,
-                 dns_search_set, provider, auth_enabled, default_os_user,
-                 web_auth_default_user=None, ca_cert_path=None):
+                 dns_search_set, provider, default_os_user, web_auth_default_user=None):
```

Comment:
because it doesnt need to be defined to create a ApiClient now. The ca_cert_path and verification is a part of the requests.session object and doesnt need to be explicitly handled until something actually wants to use that (which we can do only when security is enabled). The hook had to go here before to allow downstream (EE) to use this without having to change the request pattern with a hack